src/core/vm.c

```
@@ -1272,20 +1327,36 @@
 
       // If we reached here it's a valid callable.
       ASSERT(closure != NULL, OOPS);
 
-      // -1 argument means multiple number of args.
-      if (closure->fn->arity != -1 && closure->fn->arity != argc) {
-        char buff[STR_INT_BUFF_SIZE]; sprintf(buff, "%d", closure->fn->arity);
-        String* msg = stringFormat(vm, "Expected exactly $ argument(s) "
-                                  "for function $", buff, closure->fn->name);
-        RUNTIME_ERROR(msg);
+      // Like lua: Extra arguments are thrown away; extra parameters get null.
+      if (closure->fn->arity != -1) {
```

cppcheck

[warning] Possible null pointer dereference

(nullPointer)